tests/integration/local/start_api/test_start_api_with_terraform_application.py

```diff
@@ -99,6 +108,89 @@ def test_successful_request(self):
         self.assertEqual(response.json(), {"message": "hello world"})
 
 
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+@pytest.mark.flaky(reruns=3)
+@parameterized_class(
+    [
+        {
+            "terraform_application": "lambda-auth-openapi",
+            "expected_error_message": "Error: AWS SAM CLI is unable to process a Terraform project that uses an OpenAPI"
+            " specification to define the API Gateway resource.",
+        },
+        {
+            "terraform_application": "terraform-api-simple-multiple-resources-limitation",
+            "expected_error_message": "Error: AWS SAM CLI could not process a Terraform project that contains a source "
+            "resource that is linked to more than one destination resource.",
+        },
+        {
+            "terraform_application": "terraform-api-simple-local-variables-limitation",
+            "expected_error_message": "Error: AWS SAM CLI could not process a Terraform project that uses local "
+            "variables to define linked resources.",
+        },
+    ]
+)
+class TestStartApiTerraformApplicationLimitations(TerraformStartApiIntegrationBase):
+    @classmethod
+    def setUpClass(cls):
+        command = get_sam_command()
+        cls.command_list = [
+            command,
+            "local",
+            "start-api",
+            "--hook-name",
+            "terraform",
+            "--beta-features",
+            "-p",
+            str(random_port()),
+        ]
+        cls.test_data_path = Path(cls.get_integ_dir()) / "testdata" / "start_api"
+        cls.project_directory = cls.test_data_path / "terraform" / cls.terraform_application
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls._remove_generated_directories()
+
+    def test_unsupported_limitations(self):
+        apply_disclaimer_message = "Unresolvable attributes discovered in project, run terraform apply to resolve them."
+
+        process = self._run_command(self.command_list, check=False)
+
+        LOG.info(process.stderr)
+        output = process.stderr.decode("utf-8")
+        self.assertNotEqual(process.returncode, 0)
```

Comment:
nit: Should we assert an exit code of 1 instead of not 0? I don't think we want any of these to exit on 255, right?

Reply:
yes, let me update it

Reply:
done